Timeline sync fails with serialization exception

Hey! First of all, thank you for making this project public, I can't wait to use it daily. But I can't even get to the main timeline screen, all I see is an endless loading indicator. I looked into logcat, and this is what I see

```
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.serialization.kotlinx.KotlinxSerializationConverter.deserialize(KotlinxSerializationConverter.kt:79)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.serialization.ContentConverterKt$deserialize$$inlined$map$1$2.emit(Emitters.kt:224)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.flow.FlowKt__BuildersKt$asFlow$$inlined$unsafeFlow$3.collect(SafeCollector.common.kt:115)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.serialization.ContentConverterKt$deserialize$$inlined$map$1.collect(SafeCollector.common.kt:113)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.flow.FlowKt__ReduceKt.firstOrNull(Reduce.kt:243)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.flow.FlowKt.firstOrNull(Unknown Source:1)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.serialization.ContentConverterKt.deserialize(ContentConverter.kt:128)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.contentnegotiation.ContentNegotiation.convertResponse$ktor_client_content_negotiation(ContentNegotiation.kt:185)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.contentnegotiation.ContentNegotiation$Plugin$install$2.invokeSuspend(ContentNegotiation.kt:210)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.contentnegotiation.ContentNegotiation$Plugin$install$2.invoke(Unknown Source:13)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.contentnegotiation.ContentNegotiation$Plugin$install$2.invoke(Unknown Source:6)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:123)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.proceed(SuspendFunctionGun.kt:81)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.HttpClient$4.invokeSuspend(HttpClient.kt:173)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.HttpClient$4.invoke(Unknown Source:11)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.HttpClient$4.invoke(Unknown Source:6)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:123)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.proceed(SuspendFunctionGun.kt:81)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.logging.Logging$setupResponseLogging$2.invokeSuspend(Logging.kt:175)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.logging.Logging$setupResponseLogging$2.invoke(Unknown Source:11)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.logging.Logging$setupResponseLogging$2.invoke(Unknown Source:6)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:123)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.proceed(SuspendFunctionGun.kt:81)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.proceedWith(SuspendFunctionGun.kt:91)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.HttpCallValidator$Companion$install$2.invokeSuspend(HttpCallValidator.kt:138)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.HttpCallValidator$Companion$install$2.invoke(Unknown Source:13)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.plugins.HttpCallValidator$Companion$install$2.invoke(Unknown Source:6)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.loop(SuspendFunctionGun.kt:123)
2022-10-02 09:38:24.589 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.proceed(SuspendFunctionGun.kt:81)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.SuspendFunctionGun.execute$ktor_utils(SuspendFunctionGun.kt:101)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.util.pipeline.Pipeline.execute(Pipeline.kt:77)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.call.HttpClientCall.bodyNullable(HttpClientCall.kt:88)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.client.call.HttpClientCall.body(HttpClientCall.kt:114)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.http.ktor.internal.KtorMatrixHttpClient.execute(KtorMatrixHttpClient.kt:104)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.http.ktor.internal.KtorMatrixHttpClient.access$execute(KtorMatrixHttpClient.kt:13)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.http.ktor.internal.KtorMatrixHttpClient$execute$3.invokeSuspend(Unknown Source:15)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  Caused by: kotlinx.serialization.MissingFieldException: Field 'alias' is required for type with serial name 'app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent.CanonicalAlias.Content', but it was missing at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4]
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:90)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.Json.decodeFromString(Json.kt:95)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at io.ktor.serialization.kotlinx.KotlinxSerializationConverter.deserialize(KotlinxSerializationConverter.kt:71)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	... 43 more
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  Caused by: kotlinx.serialization.MissingFieldException: Field 'alias' is required for type with serial name 'app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent.CanonicalAlias.Content', but it was missing at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4]
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:90)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeNullableSerializableElement(AbstractDecoder.kt:79)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiSyncResponse$$serializer.deserialize(ApiSyncResponse.kt:10)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiSyncResponse$$serializer.deserialize(ApiSyncResponse.kt:10)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	... 45 more
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  Caused by: kotlinx.serialization.MissingFieldException: Field 'alias' is required for type with serial name 'app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent.CanonicalAlias.Content', but it was missing at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4]
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:90)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeNullableSerializableElement(AbstractDecoder.kt:79)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiSyncRooms$$serializer.deserialize(ApiSyncResponse.kt:36)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiSyncRooms$$serializer.deserialize(ApiSyncResponse.kt:36)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
2022-10-02 09:38:24.590 16450-19319 System.err              app.dapk.st                          W  	... 50 more
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  Caused by: kotlinx.serialization.MissingFieldException: Field 'alias' is required for type with serial name 'app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent.CanonicalAlias.Content', but it was missing at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4]
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:90)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:162)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.CompositeDecoder$DefaultImpls.decodeSerializableElement$default(Decoding.kt:533)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.MapLikeSerializer.readElement(CollectionSerializers.kt:111)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.MapLikeSerializer.readElement(CollectionSerializers.kt:84)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.AbstractCollectionSerializer.readElement$default(CollectionSerializers.kt:51)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.AbstractCollectionSerializer.merge(CollectionSerializers.kt:36)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.AbstractCollectionSerializer.deserialize(CollectionSerializers.kt:43)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	... 55 more
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  Caused by: kotlinx.serialization.MissingFieldException: Field 'alias' is required for type with serial name 'app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent.CanonicalAlias.Content', but it was missing at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4]
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:90)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:162)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiSyncRoom$$serializer.deserialize(ApiSyncResponse.kt:53)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiSyncRoom$$serializer.deserialize(ApiSyncResponse.kt:53)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	... 65 more
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  Caused by: kotlinx.serialization.MissingFieldException: Field 'alias' is required for type with serial name 'app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent.CanonicalAlias.Content', but it was missing at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4] at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4]
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:90)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:162)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiSyncRoomState$$serializer.deserialize(ApiSyncResponse.kt:91)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiSyncRoomState$$serializer.deserialize(ApiSyncResponse.kt:91)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
2022-10-02 09:38:24.591 16450-19319 System.err              app.dapk.st                          W  	... 71 more
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  Caused by: kotlinx.serialization.MissingFieldException: Field 'alias' is required for type with serial name 'app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent.CanonicalAlias.Content', but it was missing at path: $.rooms.join['RoomId(value=!PukFFdIcHgtaaHZflT:devture.com)'].state.events[4]
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:90)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:162)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.encoding.CompositeDecoder$DefaultImpls.decodeSerializableElement$default(Decoding.kt:533)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.CollectionLikeSerializer.readElement(CollectionSerializers.kt:80)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.AbstractCollectionSerializer.readElement$default(CollectionSerializers.kt:51)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.AbstractCollectionSerializer.merge(CollectionSerializers.kt:36)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.AbstractCollectionSerializer.deserialize(CollectionSerializers.kt:43)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	... 77 more
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  Caused by: kotlinx.serialization.MissingFieldException: Field 'alias' is required for type with serial name 'app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent.CanonicalAlias.Content', but it was missing
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.PluginExceptionsKt.throwMissingFieldException(PluginExceptions.kt:20)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent$CanonicalAlias$Content.<init>(ApiTimelineEvent.kt:66)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent$CanonicalAlias$Content$$serializer.deserialize(ApiTimelineEvent.kt:66)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent$CanonicalAlias$Content$$serializer.deserialize(ApiTimelineEvent.kt:66)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.PolymorphicKt.decodeSerializableValuePolymorphic(Polymorphic.kt:61)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeSerializableValue(TreeJsonDecoder.kt:52)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableValue(Tagged.kt:207)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.TaggedDecoder$decodeSerializableElement$1.invoke(Tagged.kt:280)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.TaggedDecoder.tagBlock(Tagged.kt:297)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.internal.TaggedDecoder.decodeSerializableElement(Tagged.kt:280)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent$CanonicalAlias$$serializer.deserialize(ApiTimelineEvent.kt:59)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at app.dapk.st.matrix.sync.internal.request.ApiTimelineEvent$CanonicalAlias$$serializer.deserialize(ApiTimelineEvent.kt:59)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.PolymorphicKt.decodeSerializableValuePolymorphic(Polymorphic.kt:61)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.AbstractJsonTreeDecoder.decodeSerializableValue(TreeJsonDecoder.kt:52)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.TreeJsonDecoderKt.readPolymorphicJson(TreeJsonDecoder.kt:33)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.PolymorphicKt.decodeSerializableValuePolymorphic(Polymorphic.kt:71)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:81)
2022-10-02 09:38:24.592 16450-19319 System.err              app.dapk.st                          W  	... 86 more
```

--- a/matrix/services/sync/src/main/kotlin/app/dapk/st/matrix/sync/internal/request/ApiTimelineEvent.kt
+++ b/matrix/services/sync/src/main/kotlin/app/dapk/st/matrix/sync/internal/request/ApiTimelineEvent.kt
@@ -65,7 +65,7 @@ internal sealed class ApiTimelineEvent {
 
         @Serializable
         internal data class Content(
-            @SerialName("alias") val alias: String
+            @SerialName("alias") val alias: String? = null
         )
     }
 

--- a/matrix/services/sync/src/main/kotlin/app/dapk/st/matrix/sync/internal/sync/RoomOverviewProcessor.kt
+++ b/matrix/services/sync/src/main/kotlin/app/dapk/st/matrix/sync/internal/sync/RoomOverviewProcessor.kt
@@ -66,7 +66,7 @@ internal class RoomOverviewProcessor(
 
     private suspend fun roomDisplayName(roomToProcess: RoomToProcess, combinedEvents: List<ApiTimelineEvent>): String? {
         val roomName = combinedEvents.filterIsInstance<ApiTimelineEvent.RoomName>().lastOrNull()?.content?.name
-            ?: combinedEvents.filterIsInstance<ApiTimelineEvent.CanonicalAlias>().lastOrNull()?.content?.alias
+            ?: combinedEvents.filterIsInstance<ApiTimelineEvent.CanonicalAlias>().lastOrNull()?.content?.alias?.takeIf { it.isNotEmpty() }
             ?: roomToProcess.heroes?.let {
                 roomMembersService.find(roomToProcess.roomId, it).joinToString { it.displayName ?: it.id.value }
             }

